components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/listener/WorkflowAuditLogger.java

```diff
@@ -0,0 +1,159 @@
+                + "\"";
+        AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, "Remove workflow", auditData, AUDIT_SUCCESS));
+    }
+
+    /**
+     * Trigger after adding a workflow
+     *
+     * @param workflowDTO
+     * @param parameterList
+     * @param tenantId
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostAddWorkflow(Workflow workflowDTO, List<Parameter> parameterList, int tenantId) throws
+            WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Workflow Name" + "\" : \"" + workflowDTO.getWorkflowName()
+                + "\",\"" + "Workflow Impl ID" + "\" : \"" + workflowDTO.getWorkflowImplId()
+                + "\",\"" + "Workerflow ID" + "\" : \"" + workflowDTO.getWorkflowId()
+                + "\",\"" + "Workflow Description" + "\" : \"" + workflowDTO.getWorkflowDescription()
+                + "\",\"" + "Template ID" + "\" : \"" + workflowDTO.getTemplateId()
+                + "\"";
+        AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, "Add Workflow", auditData, AUDIT_SUCCESS));
+    }
+
+    /**
+     * Trigger after adding a association
+     *
+     * @param associationName
+     * @param workflowId
+     * @param eventId
+     * @param condition
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostAddAssociation(String associationName, String workflowId, String eventId, String condition)
+            throws WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Association Name" + "\" : \"" + associationName
+                + "\",\"" + "Workflow ID" + "\" : \"" + workflowId
+                + "\",\"" + "Event ID" + "\" : \"" + eventId
+                + "\",\"" + "Condition" + "\" : \"" + condition
+                + "\"";
+        AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, "Add Association", auditData, AUDIT_SUCCESS));
+    }
+
+    /**
+     * Trigger after removing an association.
+     *
+     * @param associationId
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostRemoveAssociation(int associationId) throws WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Association ID" + "\" : \"" + associationId
+                + "\"";
```

Comment:
Thanks for pointing this, fixed with https://github.com/wso2/carbon-identity/pull/1675